lib/utils/bgpstream_utils_rpki.c

```diff
@@ -0,0 +1,210 @@
+    tmp = optarg_dup;
+    while(tmp = strstr(tmp, ",")) { count_ssh_options++; tmp++; }
+    if(count_ssh_options < MIN_SSH_CNT) {
+      fprintf(stderr, "ERROR:  Invalid RPKI SSH options\n");
+      fprintf(stderr, " * FORMAT: user,hostkey,private_key,project,collector\n");
+      inp->rpki_args_check = 0;
+      return inp;
+    }
+    free(optarg_dup);
+  }
+
+  /* Parse all SSH related input arguments */
+  if (inp->rpki_ssh && !(inp->rpki_historical)) {
+    for (int i = 0; i < RPKI_SSH_CNT; i++) {
+      strncat(inp->rpki_ssh_arg, !strlen(inp->rpki_ssh_arg) ? "" : ",",
+              sizeof(inp->rpki_ssh_arg) - 1);
+      strncat(inp->rpki_ssh_arg, strtok(NULL, ","),
+              sizeof(inp->rpki_ssh_arg) - 1);
+    }
+  }
+
+  /* Parse all project and collector arguments */
+  while ((inp->rpki_ptr = strtok(NULL, ",")) != NULL) {
+    if ((inp->rpki_options_cnt)++ % 2) {
+      strncat(inp->rpki_collectors, !strlen(inp->rpki_collectors) ? "" : ",",
+              sizeof(inp->rpki_collectors) - 1);
+      strncat(inp->rpki_collectors, inp->rpki_ptr, RPKI_CMD_CNT - 1);
+    } else {
+      strncat(inp->rpki_projects, !strlen(inp->rpki_projects) ? "" : ",",
+              sizeof(inp->rpki_projects) - 1);
+      strncat(inp->rpki_projects, inp->rpki_ptr, RPKI_CMD_CNT - 1);
+    }
+  }
+
+  inp->rpki_active = 1;
+  inp->rpki_args_check = 1;
+
+  return inp;
+}
+
+bgpstream_rpki_input_t *
+bgpstream_rpki_parse_windows(bgpstream_rpki_input_t *input,
+                             rpki_window_t windows[WINDOW_CMD_CNT],
+                             int windows_cnt) {
+
+  /* Parse all BGPReader window arguments */
+  char rpki_window[RPKI_WINDOW_LEN];
+  for (int i = 0; i < windows_cnt; i++) {
+    snprintf(rpki_window, sizeof(rpki_window), "%" PRIu32 "-%" PRIu32 ",",
+             windows[i].start, windows[i].end);
+    strcat(input->rpki_windows, rpki_window);
+  }
+  input->rpki_windows[strlen(input->rpki_windows) - 1] = '\0';
+
+  return input;
+}
+
+rpki_cfg_t *bgpstream_rpki_set_cfg(bgpstream_rpki_input_t *inp) {
+
+  /* Set up the ROAFetchlib configuration */
+  rpki_cfg_t *cfg = NULL;
+
+  return rpki_set_config(
+      inp->rpki_projects, inp->rpki_collectors, inp->rpki_windows,
+      inp->rpki_unified, inp->rpki_historical, RPKI_BROKER,
+      (inp->rpki_ssh && !inp->rpki_historical) ? inp->rpki_ssh_arg : NULL);
+}
+
+int bgpstream_rpki_validate(bgpstream_elem_t const *elem, char *result,
+                            size_t size) {
+
+  /* Validate a BGP elem with the ROAFetchlib */
+  int val_rst = 0;
+  char prefix[INET6_ADDRSTRLEN];
+  bgpstream_addr_ntop(prefix, INET6_ADDRSTRLEN,
+                      &(((bgpstream_pfx_t *)&(elem->prefix))->address));
+  uint32_t asn = 0;
+
+  if (!bgpstream_as_path_get_origin_val(elem->as_path, &asn)) {
+    if (!rpki_validate(elem->annotations.cfg, elem->annotations.timestamp, asn,
+                       prefix, elem->prefix.mask_len, result, size)) {
+      val_rst = 1;
```

Comment:
Can you add some comment here to explain the `if` statements? 

Seems to me that `rpki_validate` returns 0 if success (assuming `val_rst==1` means success), is it correct?